src/main/java/com/telefonica/iot/cygnus/backends/ckan/CKANBackendImpl.java

```diff
@@ -259,7 +262,9 @@ private void insert(String recvTime, String resourceId, Map<String, String> attr
             urlPath = "/api/3/action/datastore_upsert";
         
             // do the CKAN request
-            CKANResponse res = requester.doCKANRequest("POST", urlPath, jsonString);
+            ArrayList<Header> headers = new ArrayList<Header>();
```

Comment:
You are going to hate me probably for this :p but I saw a lot of repeated code regarding the creation of the CKAN request and URL as well as sending it :) Do you consider it appropriate to refactor this code in some private methods so it can be reused? I leave it to your discretion... ;)